R/calc_aggregate_counts.R

```diff
@@ -61,73 +61,113 @@ calc_aggregate_counts <- function(
             summarise(UCLA = sum_na_rm(UCLA), .groups = "drop")
 
+        if(collapse_vaccine){
+            sub_vac_res <- state_df %>%
+                group_by(State, Date) %>%
+                mutate(No.Initiated = !("Residents.Initiated" %in% Measure)) %>%
+                filter(No.Initiated) %>%
+                # remove vadmin in the vector if you dont want to sub for that val
+                filter(Measure %in% c("Residents.Completed", "Residents.Vadmin")) %>%
+                arrange(State, Date, Measure) %>%
+                filter(1:n() == 1) %>%
+                mutate(Measure = "Residents.Initiated") %>%
+                ungroup()
+
+            sub_vac_staff <- state_df %>%
+                group_by(State, Date) %>%
+                mutate(No.Initiated = !("Staff.Initiated" %in% Measure)) %>%
+                filter(No.Initiated) %>%
+                # add vadmin in the vector if you dont to sub for that val
+                filter(Measure %in% c("Staff.Completed", "Staff.Vadmin")) %>%
+                arrange(State, Date, Measure) %>%
+                filter(1:n() == 1) %>%
+                mutate(Measure = "Staff.Initiated") %>%
+                ungroup()
+
+            state_df <- bind_rows(state_df, sub_vac_res, sub_vac_staff) %>%
+                select(-No.Initiated)
+        }
+
         comb_df <- state_df %>%
-            full_join(mp_data, by = c("State", "Measure", "Date"))
+            full_join(mp_data, by = c("State", "Measure", "Date")) %>%
+            arrange(State, Date, Measure)
     }
     else{
-        state_df <- state_wide_df %>%
-            tidyr::pivot_longer(
-                -(Name:State), names_to = "Measure", values_to = "UCLA") %>%
+        state_df <- fac_long_df %>%
+            rename(UCLA = value) %>%
             filter(!is.na(UCLA)) %>%
             group_by(State, Measure) %>%
             mutate(has_statewide = "STATEWIDE" %in% Name) %>%
-            # if state wide and other counts exist for a measure only use statewide
+            # if state wide and other counts exist for a measure only take more
+            # recently scraped data
+            filter(!(has_statewide) | Date == max(Date)) %>%
+            mutate(has_statewide = "STATEWIDE" %in% Name) %>%
+            # if state wide and other counts still exist for a measure only
+            # use statewide measures
             filter(!(has_statewide & Name != "STATEWIDE")) %>%
             group_by(State, Measure) %>%
             summarise(
                 UCLA = sum_na_rm(UCLA), Date = max(Date), .groups = "drop")
 
+        if(collapse_vaccine){
+            sub_vac_res <- state_df %>%
+                group_by(State) %>%
+                mutate(No.Initiated = !("Residents.Initiated" %in% Measure)) %>%
+                filter(No.Initiated) %>%
+                # add vadmin in the vector if you also want to sub for that val
+                filter(Measure %in% c("Residents.Completed")) %>%
```

Comment:
Is it intentional to not collapse in `Residents.Vadmin` here but to do so in the `all_dates = TRUE` version (lines 97 and 108)? 

Don't remember whether we decided to stick with doing that vs. not doing it while we wait for responses from DOCs on what their vaccine variables mean, but I think we should we should be consistent in both places.

Reply:
Thanks for catching, has been updated